packages/actor-rdf-dereference-http-parse/lib/ActorRdfDereferenceHttpParseBase.ts

```diff
@@ -135,20 +135,25 @@ export abstract class ActorRdfDereferenceHttpParseBase extends ActorRdfDereferen
   }
 
   public mediaTypesToAcceptString(mediaTypes: { [id: string]: number }, maxLength: number): string {
-    // Ensure a ',*/*;q=0.1' suffix
-    maxLength -= 10;
-
+    const wildcard = '*/*;q=0.1';
     const parts: string[] = [];
     const sortedMediaTypes = Object.keys(mediaTypes)
       .map(mediaType => ({ mediaType, priority: mediaTypes[mediaType] }))
       .sort((left, right) => right.priority - left.priority);
-    let partsLength = 0;
+    // Take into account the ',' characters joining each type
+    const separatorLength = sortedMediaTypes.length - 1;
+    let partsLength = separatorLength;
     for (const entry of sortedMediaTypes) {
       const part = entry.mediaType + (entry.priority !== 1 ?
         `;q=${entry.priority.toFixed(3).replace(/0*$/u, '')}` :
         '');
       if (partsLength + part.length > maxLength) {
-        parts.push('*/*;q=0.1');
+        while (partsLength + wildcard.length > maxLength) {
```

Comment:
You could change this into a `do … while` since we know the condition will be true the first time.

Reply:
The condition is not necessarily true the first time. The while-condition covers the situation where there is enough space left to concatenate the wildcard, but not enough space left to concatenate a media type. Doing so anyway would result in an eager removal of a type, which would cause the original bug again.

 This scenario is covered by the first test, by the way:
![Screenshot 2020-10-02 at 13 34 10](https://user-images.githubusercontent.com/15344753/94919014-f5893880-04b3-11eb-94b3-8cfb9bc2bb31.png)

Reply:
My bad, thought the `if` and `while` had the same conditions. Good testing!